### PR TITLE
UCT/MLX5: Preparations for HW TM for PRM tls p2

### PR DESCRIPTION
**Pull request by brminich**
- Tag offload initialization/cleanup (mlx5 common)
- Some mlx5 TM infra
- XRQ ops posting for mlx5 common (RC and DC)

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5766/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5768/ for details (Mellanox internal link).

**Comment by brminich** [comment]
@yosefe, plz check

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 1]
```diff
@@ -38,16 +38,93 @@ enum {
...
+    uint8_t                       flags;
+    uint16_t                      index;
+    uint8_t                       rsvd0[2];
+    uint16_t                      sw_cnt;
+    uint8_t                       rsvd1[8];
+    uint64_t                      append_tag;
+    uint64_t                      append_mask;
+} uct_rc_mlx5_wqe_tm_seg_t;
+
+typedef struct uct_rc_mlx5_tag_entry {
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: code style: use 2 space lines between struct definitions in the header file

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 2]
```diff
@@ -38,16 +38,93 @@ enum {
...
+
+typedef struct uct_rc_mlx5_srq_op {
+    uct_rc_mlx5_tag_entry_t       *tag;
+} uct_rc_mlx5_srq_op_t;
+
+typedef struct uct_rc_mlx5_cmd_wq {
+    uct_ib_mlx5_txwq_t            super;
+    uint32_t                      qp_num;
+    uct_rc_mlx5_srq_op_t          *ops;
+    int                           ops_head;
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: pls add documentation to fields

File: src/uct/ib/rc/accel/rc_mlx5_common.c [hunk 3]
```diff
@@ -111,6 +111,87 @@ void uct_rc_mlx5_iface_common_prepost_recvs(uct_rc_iface_t *iface,
...
+    iface->tm.list = ucs_calloc(rc_iface->tm.num_tags + 1,
+                                sizeof(uct_rc_mlx5_tag_entry_t), "tm list");
+    if (iface->tm.list == NULL) {
+        ucs_free(iface->tm.cmd_wq.ops);
+        ucs_error("Failed to allocate memory for tm list");
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    for (i = 0; i < rc_iface->tm.num_tags; ++i) {
+        iface->tm.list[i].next = &iface->tm.list[i + 1];
```
[the remark this answers is not in the document]
Comment (brminich) [reply src/uct/ib/rc/accel/rc_mlx5_common.c]: we allocate `rc_iface->tm.num_tags + 1`

**Comment by yosefe** [comment]
will merge when tests pass

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5776/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5777/ for details (Mellanox internal link).

**Pull request merged**